### PR TITLE
finish button + generating message

### PR DESCRIPTION
**Pull request by rimasirich**

### CODE REVIEW

File: backend/src/yeomanui.ts [hunk 1]
```diff
@@ -189,7 +189,16 @@ export class YeomanUI {
...
-      
+	  
+	  const originalPrototype = Object.getPrototypeOf(gen);
+	  const originalGenWriting = _.get(originalPrototype, "writing");
+	  if (!originalGenWriting) {
+		originalPrototype.writing = () => { }
+	  }
+	  this.gen.on("method:writing", () => {
+		this.rpc.invoke("setInGeneratingStep", []);
+	  });
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts:200]: Is there a real need for empty func ?

	  if (originalGenWriting) {
     	       this.gen.on("method:writing", () => {
		     this.rpc.invoke("setInGeneratingStep", []);
	       });
	  }

File: backend/src/yeomanui.ts [hunk 2]
```diff
@@ -189,7 +189,18 @@ export class YeomanUI {
...
+	  const originalPrototype = Object.getPrototypeOf(gen);
+	  const originalGenWriting = _.get(originalPrototype, "writing");
+	  if (!originalGenWriting) {
+		originalPrototype.writing = () => { 
+		}
+	  }
+	  this.rpc.invoke("setInGeneratingStep", [false]);
+	  this.gen.on("method:writing", () => {
+		this.rpc.invoke("setInGeneratingStep", [true]);
+	  });
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts:202]: this. setGenWriting(gen: any)

File: frontend/src/App.vue [hunk 3]
```diff
@@ -327,12 +346,20 @@ export default {
         this.isReplaying = false;
       }
       const prompt = this.createPrompt(questions, name);
-      this.setPrompts([prompt]);
-
+	this.setPrompts([prompt]);
+	if (this.isGenerating) {
+		this.showButtons = true;
```
Comment (tomer-epstein) [inline frontend/src/App.vue:351]: true ?
shouldn't it be "this.showButtons = !this.isGenerating;"
or
if (!this.isGenerating) {

File: backend/src/yeomanui.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/yeomanui.ts]: Some gens do not have this func, but we need it here to catch it's event.
If a gen has this func we do not add nothing and use the original one.
The important part here is the event that is thrown when this func is called.

File: backend/src/yeomanui.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply backend/src/yeomanui.ts:202]: Done

File: frontend/src/App.vue (hunk 3 above)
Comment (slavik-lvovsky) [reply frontend/src/App.vue:351]: When we already in writing state buttons should not be visible, but we need to show them again when there is a need to resolve conflict with generated files. The conflict (overwrite) steps appear in after we answered on all gen questions.
We can have several screens with conflicts and should be able to press on Back and Finish.

**Pull request closed without merging**